packages/spindle-ui/src/ButtonSwitch/ButtonSwitch.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { createRef, type RefObject, useCallback, useRef } from 'react';
+import CheckBold from '../Icon/CheckBold';
+
+type Props = {
+  id?: string;
+  value: string | null;
+  options: {
+    label: string;
+    value: string;
+  }[];
+  onClick?: (value: string) => void;
+};
+
+const BLOCK_NAME = 'spui-ButtonSwitch';
+
+export const ButtonSwitch: React.FC<Props> = ({
+  id,
+  value,
+  options,
+  onClick,
+}) => {
+  const buttonsRef = useRef<RefObject<HTMLButtonElement>[]>([]);
+
+  options.forEach((_, index) => {
+    buttonsRef.current[index] = createRef<HTMLButtonElement>();
+  });
+
+  const handleKeydown = useCallback(
+    (e: React.KeyboardEvent<HTMLButtonElement>, index: number) => {
+      switch (e.key) {
+        case 'ArrowUp':
+        case 'ArrowLeft': {
+          e.preventDefault();
+          const prevButtonRef =
+            buttonsRef.current[(options.length + index - 1) % options.length];
+          const prevButton = prevButtonRef.current;
+          prevButton?.focus();
+          if (onClick) {
+            onClick(
+              options[(options.length + index - 1) % options.length].value,
+            );
+          }
+          break;
+        }
+        case 'ArrowDown':
+        case 'ArrowRight': {
+          e.preventDefault();
+          const nextButtonRef =
+            buttonsRef.current[(index + 1) % options.length];
+          const nextButton = nextButtonRef.current;
+          nextButton?.focus();
+          if (onClick) {
+            onClick(options[(index + 1) % options.length].value);
+          }
+          break;
+        }
+      }
+    },
+    [value, options.length],
```

Comment:
segmentedcontrolからとっていると思いますが、こんな感じにまとめると見通しがよくなりそうです
依存配列に使用していないvalueが入っているので消したいです
options自体も利用しているので依存配列を変更したいです
```suggestion
  const handleKeydown = useCallback(
    (e: React.KeyboardEvent<HTMLButtonElement>, index: number) => {
      const focusButton = (offset: number) => {
        const buttonRef =
          buttonsRef.current[
            (options.length + index + offset) % options.length
          ];
        const button = buttonRef.current;
        button?.focus();
        if (onClick) {
          onClick(
            options[(options.length + index + offset) % options.length].value,
          );
        }
      };

      switch (e.key) {
        case 'ArrowUp':
        case 'ArrowLeft': {
          e.preventDefault();
          focusButton(-1);
          break;
        }
        case 'ArrowDown':
        case 'ArrowRight': {
          e.preventDefault();
          focusButton(1);
          break;
        }
      }
    },
    [options, onClick],
  );
```